core/modules/widgets/widget.js

```diff
@@ -615,10 +615,11 @@ Widget.prototype.desendDestroy = function() {
 
 Widget.prototype.waitDestroy = false;
 /*
-Remove any DOM nodes created by this widget or its children
+Remove any DOM nodes created by this widget or its children, and start destroy procedure
 */
-Widget.prototype.removeChildDomNodes = function(noDestroy) {
-	// If this widget has directly created DOM nodes, delete them and exit. This assumes that any child widgets are contained within the created DOM nodes, which would normally be the case
+Widget.prototype.removeChildDomNodes = function(destroy) {
+     var destroy = (destroy !== undefined ? destroy : true);
```

Comment:
You should be able to use `destroy = destroy || true;` .. see there is no var

Reply:
would it be better to use an `options = options || {}` with an object here. It would make it possible to add parameters, without the need to change the function signature again.

Reply:
with options it would be:

```
Widget.prototype.removeChildDomNodes = function(options) {
	var options = options || {destroy:true};
	// If this widget has directly created DOM nodes, delete them and exit. This assumes that any child widgets are contained within the created DOM nodes, which would normally be the case
	if(this.domNodes.length > 0) {
		$tw.utils.each(this.domNodes,function(domNode) {
			domNode.parentNode.removeChild(domNode);
		});
		this.domNodes = [];
	} else {
		// Otherwise, ask the child widgets to delete their DOM nodes
		$tw.utils.each(this.children,function(childWidget) {
			childWidget.removeChildDomNodes({destroy:false}); 
		});
	}
	if ((options.destroy  ===  true) && !this.waitDestroy) this.desendDestroy();
}
```

I don't think any of the 3 versions are better or worse.